### DISCUSSION TITLE
Log spam

### DISCUSSION THREAD

**Discussion by YeapGuy** [ZoeyVid/NPMplus#711] [Q&A] [answered]
Now that I have NPMplus up and running well, I want to look into some of the things that are spamming my logs. I like to keep my logs as clean as possible, so that I can spot any issues ASAP.

### 1.
```
2024/03/21 14:14:29 [warn] 1278#1278: *13 an upstream response is buffered to a temporary file /usr/local/nginx/proxy_temp/3/00/0000000003 while reading upstream, client: 192.168.0.1, server: REMOVED, request: "GET /apps/memories/api/video/transcode/amj2n2vhteb0/141182/max-000000.ts HTTP/2.0", upstream: "http://172.19.0.5:11000/apps/memories/api/video/transcode/amj2n2vhteb0/141182/max-000000.ts", host: "REMOVED", referrer: "https://REMOVED/apps/memories/videos"
```
I get a huge spam of this using Nextcloud. This specific message is from the memories app, but similar messages also appear for "vanilla" Nextcloud usage (downloading/uploading files...). Is this normal or something I need to look into? If it's normal, why is it "warn"?

### 2.
```
[SSL      ] › ℹ  info      Renewing TLS certs close to expiry...
[SSL      ] › ✖  error     CommandError: Renewal configuration file /data/tls/certbot/renewal/npm-1.conf is broken. 
The error was: expected /data/tls/certbot/live/npm-1/cert.pem to be a symlink

Renewal configuration file /data/tls/certbot/renewal/npm-11.conf is broken.
The error was: expected /data/tls/certbot/live/npm-11/cert.pem to be a symlink

...etc...

0 renew failure(s), 12 parse failure(s)
    at /app/lib/utils.js:21:13
    at ChildProcess.exithandler (node:child_process:430:5)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
```

This happens every time it goes to renew TLS certs close to expiry. Doesn't happen on startup apparently.

**Comment by Zoey2936** [comment]
> I get a huge spam of this using Nextcloud. This specific message is from the memories app, but similar messages also appear for "vanilla" Nextcloud usage (downloading/uploading files...). Is this normal or something I need to look into? If it's normal, why is it "warn"?

it is normal, it is a "warn" because it slows down nginx a bit, if your upstream is fast (maybe it is localhost, in your lan or a docker container on the same host), you could try to disable proxy buffering (see compose.yaml)

> I get a huge spam of this using Nextcloud. This specific message is from the memories app, but similar messages also appear for "vanilla" Nextcloud usage (downloading/uploading files...). Is this normal or something I need to look into? If it's normal, why is it "warn"?

can you try to force renew a cert already renewed by npmplus?

**Reply by YeapGuy** [reply]
1. Okay, thanks. I've got everything on LAN, so I disabled proxy buffering. I can also do that per-host with the `proxy_buffering` directive in the Advanced tab, yes?
2. Same issue:
```
[SSL      ] › ℹ  info      Renewing Certbot certificates for Cert #11: REMOVED
[SSL      ] › ℹ  info      Command: certbot --logs-dir /tmp/certbot-log --work-dir /tmp/certbot-work --config-dir /data/tls/certbot renew --force-renewal --config "/data/tls/certbot/config.ini" --cert-name "npm-11" --preferred-challenges "dns,http" --no-random-sleep-on-renew
[Express  ] › ⚠  warning   Saving debug log to /tmp/certbot-log/letsencrypt.log
Renewal configuration file /data/tls/certbot/renewal/npm-11.conf is broken.
The error was: expected /data/tls/certbot/live/npm-11/cert.pem to be a symlink

.
0 renew failure(s), 1 parse failure(s)
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/certbot-log/letsencrypt.log or re-run Certbot with -v for more details.
```

**Reply by Zoey2936** [reply]
1. yes
2. can you do `ls -lh /opt/npm/tls/certbot/live/npm-11`?

**Reply by YeapGuy** [reply]
```
total 20K
-rwxrwx--- 1 root root  692 Mar  9 12:15 README
-rwxrwx--- 1 root root 1.6K Mar  9 12:15 cert.pem
-rwxrwx--- 1 root root 1.8K Mar  9 12:15 chain.pem
-rwxrwx--- 1 root root 3.3K Mar  9 12:15 fullchain.pem
-rwxrwx--- 1 root root  306 Mar  9 12:15 privkey.pem
```

**Reply by Zoey2936** [reply]
does your disk support symlinks?

**Reply by Zoey2936** [reply]
can you send your compose.yaml?

**Reply by YeapGuy** [reply]
I was affected by a similar issue on the original nginx-proxy-manager in the past and I used [this](https://github.com/NginxProxyManager/nginx-proxy-manager/issues/1816#issuecomment-1030841719) to resolve it. (another relevant issue: https://github.com/NginxProxyManager/nginx-proxy-manager/issues/396). My disk certainly supports symlinks.

Relevant part of my compose.yaml here. Nothing special.
```yaml
services:
  npmplus:
    image: zoeyvid/npmplus:394
    container_name: npmplus
    restart: unless-stopped
    ports:
      - '80:80'
      - '81:81'
      - '443:443'
    volumes:
      - /data/configs/npm-plus/data:/data
      #- /data/configs/npm-plus/letsencrypt:/etc/letsencrypt # not needed after migration from original NPM
    env_file:
      - stack.env
    labels:
      ...
```

**Reply by Zoey2936** [reply, marked as answer]
sorry, no Idea, but I could add the script you mentioned into npmplus

**Reply by YeapGuy** [reply]
Okay, I guess I'll just remove all the certs, restart npmplus, and regenerate them all. Quite a bit of work, but hopefully that fixes it.

**Reply by YeapGuy** [reply]
I broke my npm...
```
nginx: [emerg] cannot load certificate "/data/tls/certbot/live/npm-6/fullchain.pem": BIO_new_file() failed (SSL: error:80000002:system library::No such file or directory:calling fopen(/data/tls/certbot/live/npm-6/fullchain.pem, r) error:10000080:BIO routines::no such file)
nginx: configuration file /usr/local/nginx/conf/nginx.conf test failed
```
Seems like it only deleted the certificates partially...what do I do now?

**Reply by Zoey2936** [reply]
go to /opt/npm/nginx and delete all files inside the subfolders

**Reply by YeapGuy** [reply]
Thanks for the pointer. Instead of deleting, I just edited them manually to remove references to non-existent certificates. Now I'm back up and running. Hopefully I don't encounter this issue again. 🤞 
I don't think you need to do anything about this. It might be an issue of just the original npm.

**Reply by Zoey2936** [reply]
I've added the script you mentioned, but compressed it into two lines, while fixing a bug it had
